win32/src/win32print/win32print.cpp

```diff
@@ -2048,19 +2048,33 @@ static PyObject *PyDeviceCapabilities(PyObject *self, PyObject *args)
                 PyErr_Format(PyExc_MemoryError, "DeviceCapabilites: Unable to allocate %d bytes", bufsize);
                 break;
             }
-            ZeroMemory(buf, bufsize);
             result = DeviceCapabilities(device, port, capability, buf, pdevmode);
+            // each string in buf can be null-terminated or take up all space and have length retsize.
+            // bytes after null character can be set to junk by DeviceCapabilities().
             if (result == -1)
                 break;
             ret = PyTuple_New(result);
             if (ret == NULL)
                 break;
             retname = (TCHAR *)buf;
+            // process each returned string
             for (bufindex = 0; bufindex < result; bufindex++) {
-                if (*(retname + retsize - 1) == 0)
+                // check if returned string contains null character
+                bool contains_null = false;
```

Comment:
This could probably be simplified - it looks like `retsize_index` is always the actual string length (ie, is what we could pass to `PyWinObject_FromTCHAR` in both cases. So maybe rename it to, say, `actual_ret_size` and you can kill `contains_null` and the later `if`?

Reply:
Yep makes sense, will change.